src/spu/src/services/public/stream_fetch.rs

```diff
@@ -521,10 +458,76 @@ impl StreamFetchHandler {
 
                 debug!(read_time_ms = %now.elapsed().as_millis(),"finish sending back records");
 
-                Ok((offset.isolation(&self.isolation), true))
+                Ok((read_end_offset.isolation(&self.isolation), true))
             }
         }
     }
+
+    async fn send_processed_response(
+        &self,
+        file_partition_response: FilePartitionResponse,
+        mut next_offset: Offset,
+        batch: Batch,
+        smartstream_error: Option<SmartStreamRuntimeError>,
+    ) -> Result<(Offset, bool), SocketError> {
+        type DefaultPartitionResponse = FetchablePartitionResponse<RecordSet>;
+
+        let error_code = match smartstream_error {
+            Some(error) => ErrorCode::SmartStreamError(SmartStreamError::Runtime(error)),
+            None => file_partition_response.error_code,
+        };
+        debug!("Filter error code output: {:#?}", &error_code);
```

Comment:
either use `tracing` or implement debug trace for error code, so it can be like this:
```
debug!(error_code, "filter error code");
```